packages/reactive-element/src/css-tag.ts

```diff
@@ -166,11 +166,11 @@ export const css = (
  * shadowRoot.
  */
 export const adoptStyles = (
-  renderRoot: ShadowRoot,
+  renderRoot: Document | ShadowRoot,
```

Comment:
One thing I worry about here is that this function does no deduping. Calling it standalone like in your example in #3339 is fine, but if someone called this an override of `createRenderRoot()` they could end up appending styles for every instance of that element class.

I wonder if a better approach would be to make a `adoptDocumentStyles()` function that dedupes and calls this with `document.documentElement`.